src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Commons.cs

```diff
@@ -0,0 +1,230 @@
+﻿/*
+* Copyright (c) 2019 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+
+namespace Tizen.MachineLearning.Nnstreamer
+{
+    /// <summary>
+    /// Enumeration for pipeline state.
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
+    public enum PipelineState
+    {
+        /// <summary>
+        /// Unknown state. Maybe not constructed?
+        /// </summary>
+        Unknown = 0,
+        /// <summary>
+        /// GST-State "Null"
+        /// </summary>
+        Null = 1,
+        /// <summary>
+        /// GST-State "Ready"
+        /// </summary>
+        Ready = 2,
+        /// <summary>
+        /// GST-State "Paused"
+        /// </summary>
+        Pause = 3,
+        /// <summary>
+        /// GST-State "Playing"
+        /// </summary>
+        Playing = 4,
+    };
+
+    /// <summary>
+    /// Possible data element types of Tensor in NNStreamer.
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
+    public enum TensorType
+    {
+#pragma warning disable CA1720 // Identifier contains type name
+        /// <summary>
+        /// Integer 32bit
+        /// </summary>
+        Int32 = 0,
+        /// <summary>
+        /// Unsigned integer 32bit
+        /// </summary>
+        UInt32,
+        /// <summary>
+        /// Integer 16bit
+        /// </summary>
+        Int16,
+        /// <summary>
+        /// Unsigned integer 16bit
+        /// </summary>
+        UInt16,
+        /// <summary>
+        /// Integer 8bit
+        /// </summary>
+        Int8,
+        /// <summary>
+        /// Unsigned integer 8bit
+        /// </summary>
+        UInt8,
+        /// <summary>
+        /// Float 64bit
+        /// </summary>
+        Float64,
+        /// <summary>
+        /// Float 32bit
+        /// </summary>
+        Float32,
+        /// <summary>
+        /// Integer 64bit
+        /// </summary>
+        Int64,
+        /// <summary>
+        /// Unsigned integer 64bit
+        /// </summary>
+        UInt64,
+        /// <summary>
+        /// Unknown type
+        /// </summary>
+        Unknown,
```

Comment:
I think, Unknown is not need, 
First, if this enum is used as input, we do not allow Unknown value. 
Second, if this enum is used as output, we do not return Known, we should throw exception instead of returing Unknown

Reply:
I will.